tests/integration/models/application-membership.spec.ts

```diff
@@ -0,0 +1,371 @@
+// tslint:disable-next-line:import-blacklist
+import * as _ from 'lodash';
+import * as m from 'mochainon';
+import {
+	balena,
+	credentials,
+	givenAnApplication,
+	givenLoggedInUser,
+} from '../setup';
+import type * as BalenaSdk from '../../..';
+const { expect } = m.chai;
+import { assertDeepMatchAndLength, timeSuite } from '../../util';
+
+describe('Application Membership Model', function () {
+	timeSuite(before);
+	givenLoggedInUser(before);
+	givenAnApplication(before);
+
+	let ctx: Mocha.Context;
+	before(async function () {
+		ctx = this;
+		const roles = await balena.pine.get({
+			resource: 'application_membership_role',
+			options: { $select: ['id', 'name'] },
+		});
+		this.applicationRoleMap = _.keyBy(roles, 'name');
+		roles.forEach((role) => {
+			expect(role).to.be.an('object');
+			expect(role).to.have.property('id').that.is.a('number');
+		});
+		this.applicationDeveloperRole = this.applicationRoleMap['developer'];
+		this.applicationOperatorRole = this.applicationRoleMap['operator'];
+		this.applicationObserverRole = this.applicationRoleMap['observer'];
+		[
+			this.applicationDeveloperRole,
+			this.applicationOperatorRole,
+			this.applicationObserverRole,
+		].forEach((role) => {
+			expect(role).to.be.an('object');
+		});
+	});
+
+	describe('given a new application', function () {
+		describe('balena.models.application.membership.create()', function () {
+			before(function () {
+				ctx = this;
+			});
+
+			describe('[read operations]', function () {
```

Comment:
You only need the `ctx = this` when using `parallel`, which I think we can use here right?
```suggestion
				ctx = this;
			});

			parallel('[read operations]', function () {
```